src/server/auth/server/api_server.go

```diff
@@ -1022,18 +1062,50 @@ func (a *apiServer) getAuthenticatedUser(ctx context.Context) (*authclient.User,
 	}
 }
 
-// canonicalizeGitHubUsername corrects 'username' for case errors by looking
+// lenientCanonicalizeSubject is like 'canonicalizeUsername', except that if
+// 'subject' has no prefix, they are assumed to be a GitHub user.
+func lenientCanonicalizeSubject(ctx context.Context, subject string) (string, error) {
+	if strings.Index(subject, ":") < 0 {
+		subject = GithubPrefix + subject
+	}
+	return canonicalizeSubject(ctx, subject)
+}
+
+// canonicalizeSubject establishes the type of 'subject' by looking for one of
+// pachyderm's subject prefixes, and then canonicalizes the subject based on
+// that.
+func canonicalizeSubject(ctx context.Context, subject string) (string, error) {
+	switch {
+	case strings.HasPrefix(subject, GithubPrefix):
+		var err error
+		subject, err = canonicalizeGitHubUsername(ctx, subject[len(GithubPrefix):])
```

Comment:
So I might be misreading this code but it seems like this is basically stripping `GithubPrefix` off the front just so it can add it back to the front in `canonicalizeGitHubUsername`